src/lib/mdns/MdnsCache.h

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include <cstdint>
+
+#include <core/CHIPError.h>
+#include <core/PeerId.h>
+#include <inet/IPAddress.h>
+#include <inet/InetInterface.h>
+#include <inet/InetLayer.h>
+#include <system/SystemTimer.h>
+#include <system/TimeSource.h>
+
+// set MDNS_LOGGING to enable logging -- sometimes used in debug/test programs -- traces the behavior
+#ifdef MDNS_LOGGING
+#define MdnsLogProgress ChipLogProgress
+#else
+#define MdnsLogProgress(...)
+#endif
+
+namespace chip {
+namespace Mdns {
+
+template <size_t CACHE_SIZE>
+class MdnsCache
+{
+public:
+    MdnsCache() : elementsUsed(CACHE_SIZE)
+    {
+        nullPeerId.SetFabricId(kUndefinedFabricId);
+        nullPeerId.SetNodeId(kUndefinedNodeId);
+
+        for (MdnsCacheEntry & e : mLookupTable)
+        {
+            // each unused entry decrements the count
+            MarkEntryUnused(e);
+        }
+        MdnsLogProgress(Discovery, "construct mdns cache of size %ld", CACHE_SIZE);
+    }
+
+    // insert this entry into the cache.
+    // return error if cache is full
+    // TODO:   have an eviction policy so if the cache is full, an entry may be deleted.
+    //         One policy may be Least-time-to-live
+    CHIP_ERROR Insert(PeerId peerId, const Inet::IPAddress & addr, uint16_t port, Inet::InterfaceId iface, uint32_t TTLms)
```

Comment:
There are also other fields that need to be cached like `CRI/CRA/T` TXT fields. Maybe passing `ResolvedNodeData` here would make sense?

Reply:
ResolvedNodeData does have parsed TXT values.